packages/plugin-network-breadcrumbs/network-breadcrumbs.js

```diff
@@ -7,167 +7,165 @@ const REQUEST_METHOD_KEY = 'BS~~M'
+  let restoreFunctions = []
+  const plugin = {
+    load: client => {
+      if (!client._config.enabledBreadcrumbTypes || !includes(client._config.enabledBreadcrumbTypes, 'request')) return
+
+      const ignoredUrls = [
+        client._config.endpoints.notify,
+        client._config.endpoints.sessions
+      ].concat(_ignoredUrls)
+
+      monkeyPatchXMLHttpRequest()
+      monkeyPatchFetch()
+
+      // XMLHttpRequest monkey patch
+      function monkeyPatchXMLHttpRequest () {
+        if (!('addEventListener' in win.XMLHttpRequest.prototype)) return
+        const nativeOpen = win.XMLHttpRequest.prototype.open
+
+        // override native open()
+        win.XMLHttpRequest.prototype.open = function open (method, url) {
+          // store url and HTTP method for later
+          this[REQUEST_URL_KEY] = url
+          this[REQUEST_METHOD_KEY] = method
+
+          // if we have already setup listeners, it means open() was called twice, we need to remove
+          // the listeners and recreate them
+          if (this[REQUEST_SETUP_KEY]) {
+            this.removeEventListener('load', handleXHRLoad)
+            this.removeEventListener('error', handleXHRError)
+          }
+
+          // attach load event listener
+          this.addEventListener('load', handleXHRLoad)
+          // attach error event listener
+          this.addEventListener('error', handleXHRError)
+
+          this[REQUEST_SETUP_KEY] = true
+
+          nativeOpen.apply(this, arguments)
+        }
+
+        if (process.env.NODE_ENV !== 'production') {
+          restoreFunctions.push(() => {
+            win.XMLHttpRequest.prototype.open = nativeOpen
+          })
+        }
+      }
 
-function handleXHRLoad () {
-  if (includes(getIgnoredUrls(), this[REQUEST_URL_KEY])) {
-    // don't leave a network breadcrumb from bugsnag notify calls
-    return
-  }
-  const metadata = {
-    status: this.status,
-    request: `${this[REQUEST_METHOD_KEY]} ${this[REQUEST_URL_KEY]}`
-  }
-  if (this.status >= 400) {
-    // contacted server but got an error response
-    client.leaveBreadcrumb('XMLHttpRequest failed', metadata, BREADCRUMB_TYPE)
-  } else {
-    client.leaveBreadcrumb('XMLHttpRequest succeeded', metadata, BREADCRUMB_TYPE)
-  }
-}
+      function handleXHRLoad () {
+        if (includes(ignoredUrls, this[REQUEST_URL_KEY])) {
+          // don't leave a network breadcrumb from bugsnag notify calls
+          return
+        }
+        const metadata = {
+          status: this.status,
+          request: `${this[REQUEST_METHOD_KEY]} ${this[REQUEST_URL_KEY]}`
+        }
+        if (this.status >= 400) {
+          // contacted server but got an error response
+          client.leaveBreadcrumb('XMLHttpRequest failed', metadata, BREADCRUMB_TYPE)
+        } else {
+          client.leaveBreadcrumb('XMLHttpRequest succeeded', metadata, BREADCRUMB_TYPE)
+        }
+      }
 
-function handleXHRError () {
-  if (includes(getIgnoredUrls, this[REQUEST_URL_KEY])) {
```

Comment:
Was this a bug then?

Reply:
oh, I didn't even notice. Looks like it was!